test/Container/GeneratorFactoryTest.php

```diff
@@ -87,6 +89,33 @@ public function testSetsNamespaceFromConfig() : void
         $this->assertEquals($expected, $generator->getNamespace());
     }
 
+    public function testDefaultLogger() : void
+    {
+        $generator = (new GeneratorFactory())->create(new ServiceManager());
+        $this->assertInstanceOf(NullLogger::class, $generator->getLogger());
+    }
+
+    public function testSetsLoggerFromConfig() : void
+    {
+        $logger = $this->getMockBuilder(LoggerInterface::class)
+            ->getMockForAbstractClass();
+        $container = new ServiceManager();
+        $container->setService('MyCustomLogger', $logger);
+        $container->setService('config', [
+            'dependencies' => [
+                'auto' => [
+                    'aot' => [
+                        'logger' => 'MyCustomLogger',
+                    ],
+                ],
+            ],
+        ]);
+
+        $generator = (new GeneratorFactory())->create($container);
+        $this->assertInstanceOf(LoggerInterface::class, $generator->getLogger());
+        $this->assertNotInstanceOf(NullLogger::class, $generator->getLogger());
```

Comment:
See the info above about the `getLogger()` implementation. Can we check this differently without exposing this dependency?